keras/engine/training.py

```diff
@@ -682,6 +688,12 @@ def compile(self, optimizer, loss, metrics=[], loss_weights=None,
 
         self._collected_trainable_weights = collect_trainable_weights(self)
 
+    def _append_metric_name(self, layer_num, metric_name):
```

Comment:
Since this is ad-hoc and specific to a single method it shouldn't be a class method but rather a function defined on the fly in `compile`.